factorial/Factorial.java

```diff
@@ -0,0 +1,48 @@
+
+
+import java.util.Scanner;
+
+/**
+ * Created by buddhimah on 10/16/2017.
+ */
+public class Factorial {
+
+    public static  int fac(int number){
+
+        int r;
+
+        if(number==1){
+            return 1;
+
+        }
+        r= fac(number-1)*number;
+
+        return r;
+
+
+
+
+    }
+
+    public static void main(String args[]){
+
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Enter a number: ");
+
+        int num =sc.nextInt();
+
+        if(num<0){
+            Error a = new Error("factorial for negative numbers is undefined");
+            System.out.println(a.getMessage());
+            Factorial.main(new String[1]);
+
+
+        }else {
+
+            System.out.println(fac(num));
+        }
+
+
```

Comment:
-- extra lines.

Reply:
@t2013anurag Extra spacing and the proper spacing between operands and operators are fixed 
Thank you